beam-postgres/Database/Beam/Postgres/Connection.hs

```diff
@@ -201,7 +202,23 @@ withPgDebug dbg conn (Pg action) =
 
                    columnCount = fromIntegral $ valuesNeeded (Proxy @Postgres) (Proxy @x)
                in Pg.foldWith_ (Pg.RP (put columnCount >> ask)) conn (Pg.Query query) (PgStreamContinue nextStream) runConsumer >>= finishUp
-      step (PgRunReturning (PgCommandSyntax PgCommandTypeDataUpdateReturning syntax) mkProcess next) =
+      step (PgRunReturning AtOnce
+                           (PgCommandSyntax PgCommandTypeQuery syntax)
+                           (mkProcess :: Pg (Maybe x) -> Pg a')
+                           next) =
+        renderExecReturningList "No tuples returned to Postgres query" syntax mkProcess next
+      step (PgRunReturning _ (PgCommandSyntax PgCommandTypeDataUpdateReturning syntax) mkProcess next) =
+        renderExecReturningList "No tuples returned to Postgres update/insert returning" syntax mkProcess next
+      step (PgRunReturning _ (PgCommandSyntax _ syntax) mkProcess next) =
+        do query <- pgRenderSyntax conn syntax
+           dbg (T.unpack (decodeUtf8 query))
+           _ <- Pg.execute_ conn (Pg.Query query)
+
+           let Pg process = mkProcess (Pg (liftF (PgFetchNext id)))
+           runF process next stepReturningNone
+
+      renderExecReturningList :: (FromBackendRow Postgres x) => _ -> PgSyntax -> (Pg (Maybe x) -> Pg a') -> _ -> _
+      renderExecReturningList errMsg syntax mkProcess next =
```

Comment:
If `errMsg` isn't used, why is it here? I would think that any `*returningList` function can return an empty list without issue.